base/linalg/lq.jl

```diff
@@ -0,0 +1,115 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+# LQ Factorizations
+
+immutable LQ{T,S<:AbstractMatrix} <: Factorization{T}
+    factors::S
+    τ::Vector{T}
+    LQ(factors::AbstractMatrix{T}, τ::Vector{T}) = new(factors, τ)
+end
+
+immutable LQPackedQ{T} <: AbstractMatrix{T}
+    factors::Matrix{T}
+    τ::Vector{T}
+end
+
+LQ{T}(factors::AbstractMatrix{T}, τ::Vector{T}) = LQ{T,typeof(factors)}(factors, τ)
+
+lqfact!{T<:BlasFloat}(A::StridedMatrix{T}) = LQ(LAPACK.gelqf!(A)...)
+lqfact{T<:BlasFloat}(A::StridedMatrix{T})  = lqfact!(copy(A))
+lqfact(x::Number) = lqfact(fill(x,1,1))
+
+function lq(A::Union(Number, AbstractMatrix); thin::Bool=true)
+    F = lqfact(A)
+    F[:L], full(F[:Q], thin=thin)
+end
+
+convert{T}(::Type{LQ{T}},A::LQ) = LQ(convert(AbstractMatrix{T}, A.factors), convert(Vector{T}, A.τ))
+convert{T}(::Type{Factorization{T}}, A::LQ) = convert(LQ{T}, A)
+
+function getindex(A::LQ, d::Symbol)
+    m, n = size(A)
+    if d == :L
+        return tril!(A.factors[1:m, 1:min(m,n)])
+    elseif d == :Q
+        return LQPackedQ(A.factors,A.τ)
+    else
+        throw(KeyError(d))
+    end
+end
+
+getindex(A::LQPackedQ, i::Integer, j::Integer) = (x = zeros(eltype(A), size(A, 1)); x[i] = 1; y = zeros(eltype(A), size(A, 2)); y[j] = 1; dot(x, A*y))
```

Comment:
I know I've written many functions like this, but it should probably not be a one-liner.